datafusion/core/src/physical_plan/joins/cross_join.rs

```diff
@@ -452,10 +464,7 @@ impl CrossJoinStream {
 
                     Some(result)
                 }
-                other => {
-                    self.reservation.lock().free();
-                    other
-                }
+                other => other,
```

Comment:
My understanding is that the memory is freed on drop of the `SharedReservation`, is that correct?